### ISSUE TITLE
FacebookSDK / -ObjC flag issues

### ISSUE THREAD

**Issue by gsautr** [avivais/phonegap-parse-plugin#15]
Looks like these can be resolved if building natively but not when running this plugin. Can you take a look please?

Undefined symbols for architecture i386:
  "_FBTokenInformationExpirationDateKey", referenced from:
      -[PFFacebookTokenCachingStrategy cacheTokenInformation:] in Parse(PFFacebookTokenCachingStrategy.o)
      -[PFFacebookTokenCachingStrategy expirationDate] in Parse(PFFacebookTokenCachingStrategy.o)
      -[PFFacebookTokenCachingStrategy setExpirationDate:] in Parse(PFFacebookTokenCachingStrategy.o)
  "_FBTokenInformationTokenKey", referenced from:
      -[PFFacebookTokenCachingStrategy accessToken] in Parse(PFFacebookTokenCachingStrategy.o)
      -[PFFacebookTokenCachingStrategy setAccessToken:] in Parse(PFFacebookTokenCachingStrategy.o)
  "_FBTokenInformationUserFBIDKey", referenced from:
      -[PFFacebookTokenCachingStrategy facebookId] in Parse(PFFacebookTokenCachingStrategy.o)
      -[PFFacebookTokenCachingStrategy setFacebookId:] in Parse(PFFacebookTokenCachingStrategy.o)
  "_OBJC_CLASS_$_FBAppCall", referenced from:
      objc-class-ref in Parse(PFFacebookAuthenticationProvider.o)
  "_OBJC_CLASS_$_FBRequest", referenced from:
      objc-class-ref in Parse(PFFacebookAuthenticationProvider.o)
  "_OBJC_CLASS_$_FBSession", referenced from:
      objc-class-ref in Parse(PFFacebookAuthenticationProvider.o)
  "_OBJC_CLASS_$_FBSessionTokenCachingStrategy", referenced from:
      _OBJC_CLASS_$_PFFacebookTokenCachingStrategy in Parse(PFFacebookTokenCachingStrategy.o)
  "_OBJC_METACLASS_$_FBSessionTokenCachingStrategy", referenced from:
      _OBJC_METACLASS_$_PFFacebookTokenCachingStrategy in Parse(PFFacebookTokenCachingStrategy.o)

**Comment by hellosmithy** [comment]
+1

**Comment by sambarboza** [comment]
+1

**Comment by ianalexander** [comment]
+1

**Comment by igmt** [comment]
+1

**Comment by pieterdev** [comment]
+1. Is there any workaround? I know next to nothing about ios dev

**Comment by pskl** [comment]
Yup, same problem here. Build fails everytime, now I need to dwelve into XCode imports.

**Comment by pieterdev** [comment]
I managed to solve it by downloading the facebook sdk and adding that to the project. I guess this replaces the internal dependency in the parse api sdk? Either way, it somehow works now!

**Comment by idlework** [comment]
To solve this problem is to remove the -objc compiler flag. This way you don't have to add the Facebook framework to solve this bug.

1. Open the project in xcode
2. Navigate to Build Settings
3. Scroll down to the entry: Other Linker Flags
4. Remove the -objc entry from Other Linker Flags
5. You solve the problem by not adding extra junk to your project and removed an old compiler setting

**Comment by Tiagojdferreira** [comment]
@idlework that does solve the issue, but I am now unable to register the device using the initialize function. Did you manage to register an iOS device? If so, could you describe the procedure.

**Comment by idlework** [comment]
@Tiagojdferreira No problems here. My devices are registered correctly in Parse

The [tutorial](https://www.parse.com/tutorials/ios-push-notifications) I used to setup Parse and my implementation [gist](https://gist.github.com/idlework/eec41385b81270157eb7).

Hop that helps... let me know.

**Comment by Tiagojdferreira** [comment]
@idlework thanks, but I eventually solved the problem.

**Comment by idlework** [comment]
@Tiagojdferreira Glad to hear you solved your problem. Maybe it is good to share what your problem was and how you solved it.

**Comment by ihsanberahim** [comment]
by removing the -ObjC your cordova on ready event will never fired. That's mean your cordova not work well.

I got this problem solved.
https://github.com/ihsanberahim/phonegap-parse-plugin/blob/master/README.md

Just add the frameworks;

```
Twitter.framework
Social.framework
Accounts.framework
libsqlite3.dylib
libsqlite3.0.dylib
```

Additional Resources:
http://blog.ihsanberahim.com/2015/05/install-parse-plugin-with-ionic.html

**Comment by bluety** [comment]
@ihsanberahim thank you very much

**Comment by idlework** [comment]
@ihsanberahim That problem doesn't occur in my project. Also when I create a simple project with only this framework the ready event is dispatched.
